flair/datasets/sequence_labeling.py

```diff
@@ -443,6 +434,167 @@ def __init__(
+    ) -> None:
+        """Writes a dataset to a file.
+
+        Following these two rules:
+        (1) the text and the label(s) of every token is represented in one line separated by column_delimiter
+        (2) every sentence is separated from the previous one by an empty line
+        """
+        with open(file_path, mode="w") as output_file:
+            for sentence in _iter_dataset(dataset):
+                texts = [token.text for token in sentence.tokens]
+                texts_and_labels = [texts]
+                for label_type, level in label_type_tuples:
+                    if level is Token:
+                        texts_and_labels.append(ColumnCorpus.get_token_level_label_of_each_token(sentence, label_type))
+                    elif level is Span:
+                        texts_and_labels.append(ColumnCorpus.get_span_level_label_of_each_token(sentence, label_type))
+                    else:
+                        raise NotImplementedError(f"The level of {label_type} is neither token nor span.")
+
+                for text_and_labels_of_a_token in zip(*texts_and_labels):
+                    output_file.write(column_delimiter.join(text_and_labels_of_a_token) + "\n")
+                output_file.write("\n")
+
+    @classmethod
+    def load_corpus_with_meta_data(cls, directory: Path) -> "ColumnCorpus":
+        """Creates a ColumnCorpus instance from the directory generated by 'write_to_directory'."""
+        with open(directory / "meta_data.json") as file:
+            meta_data = json.load(file)
+
+        meta_data["column_format"] = {int(key): value for key, value in meta_data["column_format"].items()}
+
+        return cls(
+            data_folder=directory,
+            autofind_splits=True,
+            skip_first_line=False,
+            **meta_data,
+        )
+
+    def get_level_of_label(self, label_type: str) -> Union[Type[Token], Type[Span]]:
+        """Gets level of label type by checking the first label in this corpus.
+
+        Raises:
+            RuntimeError: if there is no label of label_type
+            NotImplementedError: if level of label_type is other than Token or Span
+        """
+        for dataset in [self.train, self.dev, self.test]:
+            if dataset:
+                for sentence in _iter_dataset(dataset):
+                    for label in sentence.get_labels(label_type):
+                        if isinstance(label.data_point, Token):
+                            return Token
+                        elif isinstance(label.data_point, Span):
+                            return Span
+                        else:
+                            raise NotImplementedError(
+                                f"The level of {label_type} is neither token nor span. Only token level labels and span level labels can be handled now."
+                            )
+        raise RuntimeError(f"There is no label of type {label_type} in this corpus.")
+
+    def write_corpus_meta_data(
```

Comment:
This method has only one usage, i.e. from the `write_to_directory` method. It can be indicated as an internal method by adding a "_" to the method name (`_write_corpus_meta_data`).